zjit/src/codegen.rs

```diff
@@ -1337,6 +1338,10 @@ fn gen_aref_fixnum(
     asm_ccall!(asm, rb_ary_entry, array, unboxed_idx)
 }
 
+fn gen_array_pop(asm: &mut Assembler, array: Opnd) -> lir::Opnd {
+    asm_ccall!(asm, rb_ary_pop, array)
```

Comment:
This will have to write the PC. It does allocate

You should also add a test for a frozen array because it probably also raises